01week/rockPaperScissors.js

```diff
@@ -7,27 +7,131 @@ const rl = readline.createInterface({
   output: process.stdout
 });
 
+/*
+  Whiteboarding of Rock, Paper, Scissors game:
+
+  [x] 1. User1 input of: rock, paper, or scissors
+  [x] 2. User2 input of: rock, paper, or scissors
+  [x] 3. Validate and sanitize user input
+  [x] 4. If (User1 AND User2 input is 'rock') OR (User1 AND User2 input is 'paper') OR (User1 AND User2 input is 'scissors'), it's a tie
+  [x] 5. If User1 input is 'rock' AND User2 input is 'scissors', User1 wins (rock beats scissors)
+  [x] 6. If User1 input is 'rock' AND User2 input is 'paper', User2 wins (paper beats rock)
+  [x] 7. If User1 input is 'paper' AND User2 input is 'rock', User1 wins (paper beats rock)
+  [x] 8. If User1 input is 'paper' AND User2 input is 'scissors', User2 wins (scissors beats paper)
+  [x] 9. If User1 input is 'scissors' AND User2 input is 'paper', User1 wins (scissors beats paper)
+  [x] 10. If User1 input is 'scissors' AND User2 input is 'rock', User2 wins (rock beats scissors)
+  [x] 11. Error handling in the event User1 and/or User2 input is incorrect
+*/
 
 function rockPaperScissors(hand1, hand2) {
 
-  // Write code here
+  // First we need to sanitize the user input
+  // Use toLowerCase() method to change user input to lowercase so user isn't restricted by being case-sensitive
+  // After toLowerCase method, chain on the trim() method to trim whitespace
+  const hand1LowerTrim = hand1.toLowerCase().trim();
+  const hand2LowerTrim = hand2.toLowerCase().trim();
+
+  // boolean values to hold whether or not hand1 or hand2 is correct (defaults to null, will be set in switch statements for each hand)
+  let isHand1Correct = null;
+  let isHand2Correct = null;
+
+  // check to see if hand1 has correct inputs
+  switch(hand1LowerTrim) {
+    case "rock":
+    case "paper":
+    case "scissors":
+      isHand1Correct = true;
+      break;
+    default:
+      isHand1Correct = false;
+  }
+
+  // check to see if hand2 has correct inputs
+  switch(hand2LowerTrim) {
+    case "rock":
+    case "paper":
+    case "scissors":
+      isHand2Correct = true;
+      break;
+    default:
+      isHand2Correct = false;
+  }
+
+  // if both hand1 and hand2 have correct inputs
+  if (isHand1Correct && isHand2Correct) {
+
+    // code to test who wins goes here
+
+    // If (User1 AND User2 input is 'rock') OR (User1 AND User2 input is 'paper') OR (User1 AND User2 input is 'scissors'), it's a tie
+    if((hand1LowerTrim == "rock" && hand2LowerTrim == "rock") || (hand1LowerTrim == "paper" && hand2LowerTrim == "paper") || (hand1LowerTrim == "scissors" && hand2LowerTrim == "scissors")) {
```

Comment:
this if/else block is extremely redundant, what's a better and more efficient way to solve this?

Reply:
After validating user input:

1. Check for tie
1. Check for hand1 wins
1. Check for hand2 wins